app/legacy_lib/query_supporters.rb

```diff
@@ -229,17 +229,28 @@ def self.full_filter_expr(np_id, query)
       expr = expr.and_where("payments.max_date < timezone(COALESCE(nonprofits.timezone, \'UTC\'), timezone(\'UTC\', $date)) OR payments.count = 0", date: date_ago)
     end
     if query[:search].present?
-      expr = expr.and_where(%Q(
+
+      icky_bad_search = query[:search].split.map{|s| 
```

Comment:
Do we want to keep that name? :rofl:

Reply:
Also, after the backporting you did, is this splitting still necessary?

Reply:
Probably should change then name :rofl:

Reply:
> Also, after the backporting you did, is this splitting still necessary?

It is. If you don't split, then searches like "Cacau Borges" will only match records that have Cacau Borges in the name, NOT records that have either Cacau AND/OR Borges in it.

Reply:
Name fixed!

Reply:
I tested searching for people on Facebook and Twitter, I searched for Eric Schmidt and the search didn't find you. Maybe if your name was only "Eric" in those it would come up.

Reply:
They're in the order of whatever the sort order is. The sort order default to sorting by last payment, newest first.

Reply:
Hmmm, I think this order matters.

Reply:
@clarissalimab I think, ideally, it should take similarity into account. Is that similarity consideration a must have for merging or can it be incrementally added? Or perhaps a more straightforward request: do we have an example where this limitation would lead to a significantly worse result for users?

Also, @wendylbolm, your thoughts here are important.

Reply:
I think it's worse to show results that the user is not looking for, especially on the top of the search. Imagine the case where I'm looking for a pretty common name, add the last name to make it easier to find and still get lots of results that I don't want and have to look on other pages.

What's the case that we would want to find different Erics other than the last name we're typing?

I see a similar use for that, for example when couples donate they might write "Cacau & Penelope Schultz" and "Penelope & Cacau Schultz", if the user searches for Penelope Schultz they want to see all the results involving that person, but I don't know of an example that I'd search for Penelope Schmidt and would want to see other Penelopes.

Reply:
Thoughts @wendylbolm?